samples/get_vm_names.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+# VMware vSphere Python SDK
+# Copyright (c) 2008-2015 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Python program for flat text listing the VMs on an
+ESX / vCenter, host one per line
+"""
+
+from __future__ import print_function
+import atexit
+from pyVim.connect import SmartConnectNoSSL, Disconnect
+from pyVmomi import vim
+from tools import cli
+
+
+def setup_args():
+
+    """
+    Get standard connection arguments
+    """
+    parser = cli.build_arg_parser()
+    my_args = parser.parse_args()
+
+    return cli.prompt_for_password(my_args)
+
+
+def printvminfo(vm, depth=1):
+    """
+    Print information for a particular virtual machine or recurse into a folder
+    with depth protection
+    """
+    maxdepth = 10
+
+    # if this is a group it will have children. if it does, recurse into them
+    # and then return
+    if hasattr(vm, 'childEntity'):
+        if depth > maxdepth:
+            return
+        vmlist = vm.childEntity
+        for child in vmlist:
+            printvminfo(child, depth+1)
+        return
+
+    summary = vm.summary
+    print(summary.config.name)
+
+
+def main():
+    """
+    Simple command-line program for listing the virtual machines on a host.
+    """
+
+    args = setup_args()
+    si = None
+    try:
+        si = SmartConnectNoSSL(host=args.host,
+                               user=args.user,
+                               pwd=args.password,
+                               port=int(args.port))
+        atexit.register(Disconnect, si)
+    except vim.fault.InvalidLogin:
+        pass
+
+    if not si:
+        raise SystemExit("Unable to connect to host "
+                         "with supplied credentials.")
+
+    content = si.RetrieveContent()
+    for child in content.rootFolder.childEntity:
```

Comment:
Sorry. Just realized the loop is iterating datacenters.
I was thinking checking vmFolder is kind of redundant since all datacenter should have vmFolder. But having the check doesn't hurt either.